ramips: mt7620: Adding support for Dlink DAP-1620-A2

sendurai opened the pull request:
Hello,

Adding support for D-Link DAP-1620-A2 device. 

A bit older device, but nevertheless has a decent hardware (64/8 and both 2.4 & 5). The firmware package needs to be pushed to device through U-Boot serial port and can't be done through Web UI, since the recovery occupies the second part of NAND. Can add the details to the wiki page once the code is merged and compiled.

Waiting for OpenWrt based compile of image. Anyways added the OpenWrt boot log to the wiki already. Available here:
https://openwrt.org/inbox/toh/d-link/d-link_dap-1620

Made few updates based on previous comment as noted below. Any further questions, please let me know.
https://github.com/openwrt/openwrt/pull/16911

Thanks

Commit:
ramips: mt7620: Adding support for D-link DAP-1620-A2

musashino205 commented on target/linux/ramips/dts/mt7620a_dlink_dap-1620-a2.dts:
```diff
@@ -0,0 +1,203 @@
+#include "mt7620a.dtsi"
```
Please insert `// SPDX-License-Identifier: ` with identifier(s) of your preferred license.

https://spdx.org/licenses/

musashino205 commented on target/linux/ramips/dts/mt7620a_dlink_dap-1620-a2.dts:
```diff
@@ -0,0 +1,203 @@
+				reg = <0x30000 0x10000>;
+			};
+
+			factory: partition@40000 {
```
Please remove this unused DT label (`factory: `).

musashino205 commented on target/linux/ramips/dts/mt7620a_dlink_dap-1620-a2.dts:
```diff
@@ -0,0 +1,203 @@
+				};
+			};
+
+			firmware: partition@50000 {
```
ditto (`firmware: `)

musashino205 commented on target/linux/ramips/dts/mt7620a_dlink_dap-1620-a2.dts:
```diff
@@ -0,0 +1,203 @@
+		nvmem-cell-names = "eeprom", "mac-address";
+		ieee80211-freq-limit = <5000000 6000000>;
+	};
+};
```
Please put a newline at the end of file.

raenye commented:
Did you manage to fix the RTL8211E gigabit ethernet problem?
See forum thread: https://forum.openwrt.org/t/support-for-dlink-dap-1620/72667/111

Also, you need to squash all your commits into a single commit. Never use `git merge`, only `git rebase`.

sendurai commented:
> Did you manage to fix the RTL8211E gigabit ethernet problem? See forum thread: https://forum.openwrt.org/t/support-for-dlink-dap-1620/72667/111
> 
> Also, you need to squash all your commits into a single commit. Never use `git merge`, only `git rebase`.

Just to highlight this is Revision A hardware. (A1/A2). (mt7620). The discussion thread seems to focus on B hardware?

Yes, the 1G speed works. Luci image attached. 
![DAP-1620-A2-Switch](https://github.com/user-attachments/assets/9f1bf007-13eb-4b52-a00e-277a204d3bf1)

sendurai commented:
Commits merged into single commit and ready for review/merge. Thanks

musashino205 commented:
Please use `Add` instead of `Adding` for the commit title.

https://openwrt.org/submitting-patches#submission_guidelines

> - write commit subject and message in the imperative form: “add support for X”, NOT “added support for X”

raenye commented:
> The discussion thread seems to focus on B hardware?

Actually no, the vast majority of the discussion is about rev A. Only the last couple of posts are about rev B.
Perhaps you should quickly go over the thread and see whether anything there is still relevant.

sendurai committed:
Revert "ramips: mt7620: add support for Dlink DAP-1620-A2"

This reverts commit fa68b42926761f060aea8fe36800b7c000a2a9bc.

sendurai committed:
ramips: mt7620: add support for Dlink DAP-1620-A2

Add support for DAP-1620 Revision A hardware. The original firmware has dual image, one primary and another for recovery. This version supports using only single image due to size limitations. Recovery and restore can be done using the U-Boot menu on device startup (serial port) and writing the image using option 2. Full backup of the Nand is recommended before installing in case the original partitions are to be restored in future. Full details will be updated in device wiki page.

sendurai commented:
Sorry. Will reopen another pull request. Had to delete my local branch and cannot reopen another request here.

Neustradamus commented:
Replaces:
- https://github.com/openwrt/openwrt/pull/16911

Replaced by:
- https://github.com/openwrt/openwrt/pull/17018
- https://github.com/openwrt/openwrt/pull/17020
- https://github.com/openwrt/openwrt/pull/17027

sendurai commented:
Since I deleted that branch and created one was not able to re-open. This is available in a new PR.
https://github.com/openwrt/openwrt/pull/17027
